internal/analytics/client/sync/client.go

```diff
@@ -125,6 +127,14 @@ func (a *Client) Wait() {
 
 // Events returns a channel to feed eventData directly to the report loop
 func (a *Client) report(category, action, label string, dimensions *dimensions.Values) {
+	send := true
+	if a.cfg != nil && a.cfg.IsSet(constants.ReportAnalayticsConfig) {
+		send = a.cfg.GetBool(constants.ReportAnalayticsConfig)
+	}
+	if !send {
+		return
+	}
```

Comment:
```suggestion
	if a.cfg != nil && a.cfg.IsSet(constants.ReportAnalayticsConfig) && a.cfg.GetBool(constants.ReportAnalayticsConfig) {
		return
	}
```

Reply:
Also; is there a use-case for `a.cfg` being nil? I think that constitutes an error in our code and I'd almost rather want it to bubble up to the panic/recover so that we'd hear about it rather than leave use flying blind.